usr/src/uts/common/fs/zfs/zfs_dir.c

```diff
@@ -774,25 +789,49 @@ zfs_link_create(zfs_dirlock_t *dl, znode_t *zp, dmu_tx_t *tx, int flag)
 	return (0);
 }
 
+/*
+ * The match type in the code for this function should conform to:
+ *
+ * ------------------------------------------------------------------------
+ * fs type  | z_norm      | lookup type | match type
+ * ---------|-------------|-------------|----------------------------------
+ * CS !norm | 0           |           0 | exact
+ * CS  norm | formD       |           0 | MT_IGNORE_FORM
```

Comment:
Why is this `formD`, as opposed to using the `normalization` property (indicated here by `formX`)?